src/Fmt_ast.ml

```diff
@@ -348,7 +357,20 @@ let fmt_constant c ~loc ?epi const =
           | Some s -> (s, `Preserve) )
         | _ -> (s, c.conf.escape_strings)
       in
+      let contains_pp_commands =
+        let is_substring substring = String.is_substring s ~substring in
+        List.exists ["@,"; "@;"; "@\\n"] ~f:is_substring
```

Comment:
You could avoid calculating `contains_pp_commands` by checking if the `break_on_pp_commands` logic below changed the string.

Reply:
Tried to do that, but failed because the lists are not the same in the check and the substitution. If we find a better way to write that I would be happy about it, otherwise I think we can just merge.